From: Benjamin Berg <bberg@redhat.com>
Date: Fri, 20 Nov 2020 11:23:08 +0100
Subject: usb-protection: Do not warn about ServiceUnknown errors for USBGuard

It is expected for USBGuard to not be installed on many systems. We
should not warn about that DBus error, as it does not indicate an issue.

Bug: https://gitlab.gnome.org/GNOME/gnome-settings-daemon/-/issues/567
Origin: upstream, 3.38.2, commit:affebf890fdc032f246490f6bcd8f2c7777c7be1
---
 plugins/usb-protection/gsd-usb-protection-manager.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/plugins/usb-protection/gsd-usb-protection-manager.c b/plugins/usb-protection/gsd-usb-protection-manager.c
index 0b0557f..63a4e0d 100644
--- a/plugins/usb-protection/gsd-usb-protection-manager.c
+++ b/plugins/usb-protection/gsd-usb-protection-manager.c
@@ -154,7 +154,8 @@ dbus_call_log_error (GObject      *source_object,
         result = g_dbus_proxy_call_finish (G_DBUS_PROXY (source_object),
                                            res,
                                            &error);
-        if (result == NULL)
+        if (result == NULL &&
+            !g_error_matches (error, G_DBUS_ERROR, G_DBUS_ERROR_SERVICE_UNKNOWN))
                 g_warning ("%s: %s", msg, error->message);
 }
 
